bin/metadata.js

```diff
@@ -68,3 +68,13 @@ main(PR_ID, OWNER, REPO).catch((err) => {
   logger.error(err);
   process.exit(-1);
 });
+
+function parsePRId(id) {
+  // Fast path: numeric string
+  if (`${+id}` === id)
+    return +id;
+  const match = id.match(/^https:.*\/([0-9]+)$/);
```

Comment:
perhaps try `new URL(id)`, and fallback if it errors?

Reply:
yeah, question: fall back to what? :smile:

Reply:
heh... good point ;-) ... try `new url.URL(id)`, check the protocol, and error if it's not a valid https URL

Reply:
Maybe a `\/?` at the end because sometimes people do paste the slash, like `https://github.com/nodejs/node/pull/14782/`...also maybe we can just make the regexp out of `repo` and `owner`? That way we can validate it's a pull instead of an issue as well (right now if it's an issue we just throw the errors around).

Reply:
Ah wait, we don't provide command line options for `repo`/`owner` right now :/ OK maybe just validate `pull/([0-9])+\/?` part..

Reply:
Yup – I’ve added checks for `pull/` and `/` or `/files` at the end :)